### ISSUE TITLE
Progress jumps to 84%, then does not progress for a long time, then jumps to 100%

### ISSUE THREAD

**Issue by eyalroz** [cizordj/molotov#1]
I've tried running molotov to burn a certain ISO. This is the console output:
```
Umounting /dev/sdd and /path/to/the.iso
Removing /tmp/tmp.VwB7JENmGT and /tmp/tmp.6Rko9YGEGb
Removing file system signature from /dev/sdd
Writing new partition table for /dev/sdd
Partition #1 contains a ntfs signature.
Formatting /dev/sdd1 with NTFS
Mounting /dev/sdd1 at /tmp/tmp.6Rko9YGEGb
Mounting /path/to/the.iso at /tmp/tmp.VwB7JENmGT
mount: /tmp/tmp.VwB7JENmGT: WARNING: source write-protected, mounted read-only.
Copying iso contents to /tmp/tmp.6Rko9YGEGb/
```
The I see a percentage start to climb. It climbs to 84% very quickly - within a few seconds. Then it sits at 84% for... maybe 5 or 10 minutes, I didn't time this exactly. Then it goes to 100%. Then it installs grub, which for some reason takes a lot of time - which is not reflected in any percentage/progress count. At some point I gave up and came here to write this bug report...

Edit: ... eventually the grub-install concluded successfully, but it took a _lot_ of time.

So, bottom line: Progress reporting needs to improve in terms of all of update resolution, scope and accuracy.

**Comment by cizordj** [comment]
Hi, thanks for reporting this. Let me explain this to you how the algorithm woks, it's pretty dumb to be honest.

The progress is based on the number of files that were copied from the ISO image and there is one big file that takes a lot more to copy and it makes it look like it's stuck in 84%, this happens to me as well but I didn't think it was an issue.

In essence, the ISO image has a lot of small files that are copied very quickly and that's why it goes straight up to 80% and then there's this one big file that takes a lot more to copy and once it finishes the molotov shows a 100% because the rest of the files are small too.

The reason it's taking long to install grub (or even copy the files) is that you are probably using an USB 2.0 drive, those are very slow and as stated in the project's [README](https://github.com/cizordj/molotov#known-limitations) it may take up to ten minutes to boot.

The speed it takes to install grub and copy the files is not something I can control, but I do can improve the percentage progress to make it more reliable without guarantees.

**Comment by eyalroz** [comment]
> but I do can improve the percentage progress to make it more reliable without guarantees.

Yes, that's what I'm asking. If you could do time estimations as the basis fo percentile estimation, that would help; if you can't, and can at least indicate that you're giving the percentage of files copied - that's already an improvement

**Comment by cizordj** [comment]
Okay then, that's planned for the next release, I'll keep this issue open.
